python/pyspark/sql/connect/plan.py

```diff
@@ -111,6 +112,46 @@ def _child_repr_(self) -> str:
         return self._child._repr_html_() if self._child is not None else ""
 
 
+class DataSource(LogicalPlan):
```

Comment:
is it generated by proto?

Reply:
This is not. This class is more or less the same as the DSL that we introduced for Scala: the core idea is to provide a way for `toProto`.